module/zfs/metaslab.c

```diff
@@ -2893,6 +2900,26 @@ metaslab_set_fragmentation(metaslab_t *msp, boolean_t nodirty)
 	msp->ms_fragmentation = fragmentation;
 }
 
+/*
+ * A dRAID metaslabs ms_start is always stripe aligned, this causes their sizes
+ * to vary by a few sectors. The block allocator may get confused and pick a
+ * distant metaslab because the closer ones are slightly smaller. The small
```

Comment:
Yes, that's the intent.  In practice, I doubt it's very important to fill the metaslabs initially in order.   There isn't any critical case I'm aware of where this would really matter.  This change was part of the original dRAID PR and since there may be a case I haven't considered I opted to keep it.  If you feel strongly about it it's something we could consider dropping.

Reply:
To me, it seems hard to believe that the initial fill order would really matter.  And if it did, I'd prefer that we make it more explicit, like if there are any untouched metaslabs, we explicitly select the one with the lowest ID.  So I'd prefer to drop this change, lest we drag it along for decades without knowing why :-)

Reply:
Sounds good.  Done.